app/grandchallenge/participants/views.py

```diff
@@ -94,8 +98,48 @@ def get_queryset(self):
         queryset = queryset.filter(
             Q(challenge=self.request.challenge)
         ).select_related("user__user_profile", "user__verification")
+
         return queryset
 
+    def get_context_data(self, **kwargs) -> dict[str, Any]:
+        context_data = super().get_context_data(**kwargs)
+
+        questions = self._get_registration_questions()
+        context_data["registration_questions"] = questions
+
+        answers = self._get_registration_questions_answers(questions=questions)
+
+        answer_lookup = defaultdict(dict)
+        for answer in answers:
+            answer_lookup[answer.registration_request.pk][
+                answer.question.pk
+            ] = answer
+
+        context_data["registration_questions_answer_lookup"] = answer_lookup
```

Comment:
This feels like the wrong approach. Why isn't a `prefetch_related` being used and the answers then being accessed via `RegistrationRequest.questions.answers`?

Reply:
Done!